src/createAgentManager.ts

```diff
@@ -107,8 +107,11 @@ function initializeStreamAndChat(
     return new Promise<{ chat?: Chat; streamingManager: StreamingManager<CreateStreamOptions> }>(
         async (outerResolve, outerReject) => {
             messageSentTimestamp = 0;
+            let streamingManager: StreamingManager<CreateStreamOptions> | null = null;
+
+            const timeoutId = setTimeout(async () => {
```

Comment:
move this `setTimeout` call below `createStreamingManager` call
1. You will start the timeout only after creating the manager (as opposed to now - where you wait 30 sec from the call to create manager, instead of 30 sec from the actual creation time)
1. You won't have to declare `let streamingManager...`